### ISSUE TITLE
randoop_daikon.py failed, dljc complained unrecognized arguments: --dyntrace-libs

### ISSUE THREAD

**Issue by XujieSi** [aas-integration/integration-test2#34]
Got an error (detailed log is attached) from dljc when running randoop_daikon.py.
It might be caused by the incorrect version of dljc. The one used is the master branch of  https://github.com/SRI-CSL/do-like-javac

[log_randoop_daikon.txt](https://github.com/aas-integration/integration-test2/files/724673/log_randoop_daikon.txt)

**Comment by tim-mccarthy** [comment]
The problem is actually that randoop_daikon.py is quite outdated, rather than that dljc is the wrong version.

@martinschaef Do we want to update randoop_daikon.py or just remove it?

**Comment by XujieSi** [comment]
@tim-mccarthy I am trying to get some "reasonable" regression tests using Randoop.  But I have no experience of configuring it, so I started using randoop_daikon.py.  I guess Randoop is still used somehow in other scripts, if not randoop_daikon.py. 

Could you please point out some other scripts running randoop I can leverage?

**Comment by bjkeller** [comment]
I use run_dyntrace.py when I'm testing Randoop on the corpus. As of Friday, dljc has better Randoop defaults, but it runs longer. I'm happy to help with this if you need it. 
Ben 

--
Ben Keller 
bjkeller@vt.edu

**Comment by martinschaef** [comment]
@XujieSi run_dyntrace should be the way to go. 
@tim-mccarthy could you double check if ` randoop_daikon.py` interferes with `run_dyntract`? If not, we can delete it. I'm pretty sure we don't need it.

**Comment by tim-mccarthy** [comment]
I think it should be safe to delete randoop_daikon.py. It looks like run_dyntrace.py does the same thing, but correctly. randoop_daikon.py is probably a carryover from integration-test v1.

**Comment by martinschaef** [comment]
@tim-mccarthy then lets remove it.

@XujieSi, let me know if the ` run_dyntrace.py` script works for you.

**Comment by XujieSi** [comment]
@martinschaef run_dyntrace.py  failed to run on boofcv, but seems fine on others like Sort*.

[build_output.txt](https://github.com/aas-integration/integration-test2/files/727333/build_output.txt)
[log_run_dyntrace.txt](https://github.com/aas-integration/integration-test2/files/727334/log_run_dyntrace.txt)

**Comment by martinschaef** [comment]
@bjkeller could you take it from here? I can't see what's going on from the log files. Have you run randoop on boofcv?

**Comment by bjkeller** [comment]
Several of the libraries have failures of different kinds. It will be later today or tomorrow before I can say more. 

--
Ben Keller 
bjkeller@vt.edu

**Comment by martinschaef** [comment]
@XujieSi did you make any progress today? Otherwise we can have a call tomorrow. Tim will be at the NY office and we can do live debugging.

**Comment by bjkeller** [comment]
OK, I thought the problem was related to Randoop.  The log @XujieSi posted shows boofcv failing to compile.  I have not had any compilation problems.

I posted new coverage numbers the middle of last week when I ran Randoop with the current larger bounds in `dyntrace.py` in [do-like-javac](https://github.com/SRI-CSL/do-like-javac), but have had to be away since then and haven't been able to look at the logs to see what happened.  The longer Randoop runs that I had done in December mostly terminated normally, with I think the only odd runs being from code that calls `System.exit(1)` when there is an error.  The other problems occur when the tests are run from the command line, and mostly occur because the tests are flaky – there is some global state that is maintained within Randoop, but lost when the test is run from the command line.  These are all things that we are looking forward to developing solutions to.

I can be more useful when I get back to the office, which will probably be no later than Monday.
